### PR TITLE
Refactor dropdown's hide functionality

### PR DESCRIPTION
**Pull request by rohit2sharma95**
Share the same code between `hide` and `clearMenus` methods.
Ref: https://github.com/twbs/bootstrap/pull/33451#discussion_r600059974

### CODE REVIEW

File: js/src/dropdown.js [hunk 1]
```diff
@@ -263,6 +242,30 @@ class Dropdown extends BaseComponent {
     })
   }
 
+  _completeHide(element, menu, relatedTarget) {
+    const hideEvent = EventHandler.trigger(element, EVENT_HIDE, relatedTarget)
```
Comment (GeoSot) [inline js/src/dropdown.js:246]: May be mistaken, but as I can see this is a private method.
could be changed to `_completeHide(relatedTarget)`
as `element = this._element` and `menu=this._menu`

File: js/src/dropdown.js [hunk 2]
```diff
@@ -448,29 +447,13 @@ class Dropdown extends BaseComponent {
         if (event.type === 'keyup' && event.key === TAB_KEY && dropdownMenu.contains(event.target)) {
```
Comment (GeoSot) [inline js/src/dropdown.js:447]: `dropdownMenu` can be `context._menu` and remove line 431

File: js/src/dropdown.js [hunk 3]
```diff
@@ -438,6 +433,10 @@ class Dropdown extends BaseComponent {
         continue
```
Comment (GeoSot) [inline js/src/dropdown.js:433]: `toggles[i]` can be changed to  `context._element`

File: js/src/dropdown.js (hunk 1 above)
Comment (rohit2sharma95) [reply js/src/dropdown.js:246]: True :smile:

**Review by GeoSot — APPROVED** [review]

**Pull request merged**